sdk/v2/core/events.go

```diff
@@ -197,10 +197,14 @@ type GitDetails struct {
 	// CloneURL specifies the location from where a source code repository may
 	// be cloned.
 	CloneURL string `json:"cloneURL,omitempty"`
-	// Commit specifies a commit (by sha) to be checked out.
+	// Commit specifies a revision (by SHA) to be checked out. If non-empty, this
+	// field takes precedence over any value in the Ref field.
 	Commit string `json:"commit,omitempty"`
-	// Ref specifies a tag or branch to be checked out. If left blank, this will
-	// default to "master" at runtime.
+	// Ref is a symbolic reference to a revision to be checked out. If non-empty,
+	// the value of the Commit field supercedes any value in this field. Example
+	// uses of this field include referencing a branch (refs/heads/<branch name>)
+	// or a tag (refs/tags/<tag name>). If left blank, this field is interpreted
+	// as a reference to the repository's default branch.
```

Comment:
Saw an opportunity here to improve these comments further while I was already touching them. Did the same in a few other places as well.